### ISSUE TITLE
Increase error-tolerance for searches

### ISSUE THREAD

**Issue by trugwaldsaenger** [hbz/oerworldmap#605]
The search should provide right results, even if the search input was spelled wrongly. 
One example:
"Le Mill" does not provide any results. "LeMill" provides the right result (https://oerworldmap.org/resource/urn:uuid:c407eede-7f00-11e5-a636-c48e8ff4ea31). I guess that there will be other examples similar to this one, maybe we could check if there exists a checklist for "frequent mistakes" or something like this.

**Comment by trugwaldsaenger** [comment]
Here comes another example: When I search for "unglue" I receive not hits. When I search for "unglue.it" or "unglue*" the right result (https://oerworldmap.org/resource/?q=unglue*) is found.

**Comment by philboeselager** [comment]
@literarymachine Is there something already included in non-master branches?

**Comment by literarymachine** [comment]
@philboeselager No, I don't think so. But once #528 is live, we have more freedom to experiment with elasticsearch configuration!

**Comment by philboeselager** [comment]
@literarymachine just a side note: while trying to run BaseRepositoryTest as JUnit test from within Eclipse, [this akka.actor.Props error](https://groups.google.com/forum/#!topic/akka-user/YVYTYTdVRLQ) occurred. While it is already known for 2 years, it is hopefully quite now fixed, see [this issue](https://github.com/akka/akka/issues/16969) and [this pull request](https://github.com/akka/akka/pull/20488). After the fix will have been released, I would like to update our akka dependency to the newest release. Would that be fine for you?

**Comment by literarymachine** [comment]
@philboeselager As far as I know, the Play framework altogether is a big akka actor system. We thus don't have an explicit dependency in our build.sbt. Should we maybe rather think about updating our Play version?

**Comment by philboeselager** [comment]
Yes, soon.

**Comment by philboeselager** [comment]
The solution to this issue should be using another Elasticsearch analyzer for the affected fields. Analyzers can be quite expensive in terms of performance, especially on storage where currently there is already [a performance issue](https://github.com/hbz/oerworldmap/issues/528#issuecomment-218110333).

@trugwaldsaenger Keeping performance in mind, which fields do you wish to be searchable with the flexibility described by your examples given above? For example "name" and "alternateName"?

**Comment by philboeselager** [comment]
I've tried to use different analyzers - with no success up to now. I've asked a [question](http://stackoverflow.com/questions/37190182/elasticsearch-type-multi-field-mapping-switched-to-index-no-type) - but no solution answer up to now. There's one guy asking if we maybe use different mappings in combination - but we don't do that. @literarymachine What do you think is the best thing to do now?

**Comment by philboeselager** [comment]
Ping @trugwaldsaenger (please see question above).

**Comment by literarymachine** [comment]
@philboeselager maybe you should have a chat with @jprante: http://jprante.github.io/lessons/2012/05/16/Multilingual-analysis-for-title-search.html

**Comment by trugwaldsaenger** [comment]
> @trugwaldsaenger Keeping performance in mind, which fields do you wish to be searchable with the flexibility described by your examples given above? For example "name" and "alternateName"?

I guess that the "name and the "alternateName" will make a good start. Additional the fields "description" and "tags" could be intersting as well. But I`m not sure about this....

**Comment by literarymachine** [comment]
> Additional the fields "description" and "tags" could be intersting as well.

I agree that `description` makes a lot of sense, actually any field that contains natural language. `tags` however are just that, literal tags, so I tend to not include them.

**Comment by trugwaldsaenger** [comment]
Another relevant example. OpenStax is one of the most popular textbook providers in the US. If I search for "Stax" I receive sero hits. I have to type in "openstax" to find it...

**Comment by trugwaldsaenger** [comment]
Any news on this issue? We should solve this soon, otherwise an increase number of authors might cause dublicates...

**Comment by philboeselager** [comment]
I'm sorry, no hot news, yet. The problem is: 

- I can't develop these functional parts on Eclipse, since it is blocked by an akka bug. (akka was introduced while implementing the triple store. @literarymachine didn't suffer from that problem since he doesn't use Eclipse.)
- So I decided to switch to IntelliJ, "the other" big IDE. While this is generally possible, I'm still trying to solve debugging / testing problems (see #706) for quite a big number of tests are still failing there. We do not have a definition on that yet, but my personal opinion is that software has to be developable on at least one established IDE, because a debugger is an essential tool for (backend) developing. (We could try other IDEs as well, but that might not be a model strategy.)

I'm quite messed up by these problems. They were introduced by an unlucky combination of components. Though I had no participation in creating this combination, I am the one who has to solve the resulting problem, because I am the only developer using Eclipse or IntelliJ for backend development and testing. I can't do anything more than try and try and try.

As progress is very slow on #706 and at least the tests for #605 are executable on IntelliJ, I am going to work on this issue for now and leave #706 unsolved for now.

**Comment by literarymachine** [comment]
> "Le Mill" does not provide any results. "LeMill" provides the right result 

Fixed by #752: https://oerworldmap.org/resource/?q=Le+Mill

> Here comes another example: When I search for "unglue" I receive not hits.

Fixed by #752: https://oerworldmap.org/resource/?q=unglue

**Comment by acka47** [comment]
The first two example wqueries provided by @trugwaldsaenger work, the third one doesn't though: https://oerworldmap.org/resource/?q=stax

**Comment by literarymachine** [comment]
@philboeselager probably the solution in https://github.com/hbz/oerworldmap/issues/651#issuecomment-229616871 was a bit to simple for this; searching for `*stax` works, but prepending a `*` to all queries probably breaks existing functionality that use the elasticsearch query syntax.

**Comment by literarymachine** [comment]
> The first two example wqueries provided by @trugwaldsaenger work, the third one doesn't though: https://oerworldmap.org/resource/?q=stax

Closing since this is rather a variation of #651.

**Comment by trugwaldsaenger** [comment]
@philboeselager : Here comes another example: Looking for "GOGN" does not receive any results, since the alternate name of the Global OER Graduates Network seems to be "GO-GN".

**Comment by philboeselager** [comment]
@trugwaldsaenger We should be able to realize that by the use of a RegEx pattern replace mechanism, similar to the one described [here](http://stackoverflow.com/questions/38981165/tokenize-a-string-in-elasticsearch) for apostrophes (also side note to myself).

**Comment by philboeselager** [comment]
Tested by https://github.com/hbz/oerworldmap/pull/972/commits/16e9932b6612205703478d1ce3959162678e912d
